src/Compilers/CSharp/Portable/Symbols/Source/SourceAssemblySymbol.cs

```diff
@@ -378,6 +382,8 @@ private Version AssemblyVersionAttributeSetting
             }
         }
 
+        public override Version AssemblyVersionInSource => AssemblyVersionAttributeSetting ?? Identity.Version;
```

Comment:
Maybe I should rename the method. For PE I was thinking we only emit an attribute if the version specified in source *. If the version specified in source doesn't have * we infer that it was specified exactly as is in AssemblyDef table.